core/os/memory.h

```diff
@@ -41,7 +41,6 @@
 #endif
 
 class Memory {
-	Memory();
```

Comment:
If namespaces are really, really banned, I'd go for the comprehensive deletion of compiler generated functions, maybe with a macro.

Reply:
Something like a GODOT_DELETE_GENERATED(Memory); ?

Reply:
Yes. Regarding naming, I'd go for something totally explicit like `FORBID_CONSTRUCT_AND_COPY(m_class)` or conceptual like `FUNCTION_COLLECTION(m_class)` or `PSEUDONAMESPACE(m_class)`. Not very convinced about these names either, but that'd be the idea.

Reply:
> Or stop abusing classes to group free functions and use namespaces instead. Godot is not using namespaces, but I'd rather use them as an exception in these cases rather than adding so much boilerplate.

I think namespaces for this can make sense, let's have a discussion about it on `#core` :)

Reply:
For this PR's sake (and the `master` counterpart which was already merged), I would suggest going for the `= delete` approach. We can always refactor further in a different PR after reaching consensus.